conn.go

```diff
@@ -415,20 +422,22 @@ func (c *Conn) identify() (*IdentifyResponse, error) {
 }
 
 func (c *Conn) upgradeTLS(tlsConf *tls.Config) error {
-	host, _, err := net.SplitHostPort(c.addr)
-	if err != nil {
-		return err
-	}
 
 	// create a local copy of the config to set ServerName for this connection
 	conf := &tls.Config{}
 	if tlsConf != nil {
 		conf = tlsConf.Clone()
 	}
-	conf.ServerName = host
+	if c.socketType == "tcp" {
```

Comment:
You did !unixSocket in another place. Maybe it would be better for consistency to do the same here.

Reply:
also, I don't think we will ever get in here (inside upgradeTLS) for unixSocket. How / why would we?

Reply:
Keeping backward compatibility with TLS and no need to specify conflicting options explicitly